serve.json

```diff
@@ -3,5 +3,55 @@
   "trailingSlash": false,
   "rewrites": [
     { "source": "/certification/**", "destination": "/certification/index.html" }
+  ],
+  "redirects": [
+    {
+      "source": "/learn/apis-and-microservices",
+      "destination": "/learn/back-end-development-and-apis"
+    },
+    {
+      "source": "/learn/apis-and-microservices/apis-and-microservices-projects",
+      "destination": "/learn/back-end-development-and-apis/back-end-development-and-apis-projects"
+    },
+    {
+      "source": "/learn/apis-and-microservices/:block",
+      "destination": "/learn/back-end-development-and-apis/:block"
+    },
+    {
+      "source": "/learn/apis-and-microservices/apis-and-microservices-projects/:challenge",
+      "destination": "/learn/back-end-development-and-apis/back-end-development-and-apis-projects/:challenge"
+    },
+    {
+      "source": "/learn/apis-and-microservices/:block/:challenge",
+      "destination": "/learn/back-end-development-and-apis/:block/:challenge"
+    },
+    {
+      "source": "/certification/:username/apis-and-microservices",
+      "destination": "/certification/:username/back-end-development-and-apis"
+    },
+    {
+      "source": "/learn/front-end-libraries",
+      "destination": "/learn/front-end-development-libraries"
+    },
+    {
+      "source": "/learn/front-end-libraries/front-end-libraries-projects",
+      "destination": "/learn/front-end-development-libraries/front-end-development-libraries-projects"
+    },
+    {
+      "source": "/learn/front-end-libraries/:block",
+      "destination": "/learn/front-end-development-libraries/:block"
```

Comment:
Shouldn't this one come later then? It's less specific than ones below it.

Reply:
Good question.  Order only seems to matter when the number of path segments match. So, since all the following sources are longer (or completely different) it doesn't matter.

Reply:
There's definitely some magic going on there that I'd worry about breaking later lol

Reply:
It's fairly light magic, tbh.  It just checks a path against the array of redirects and uses the first redirect that matches.  The only subtlety is that if two redirects could match a given path, you have to put the redirect you want first.

Also, I've added a bunch of tests to the CI, so we should find out if this breaks.